### DISCUSSION TITLE
Can this be used to hide the microphone icon?

### DISCUSSION THREAD

**Discussion by connerdassen** [valinet/ExplorerPatcher#1867] [Q&A]
I use VoiceMeeter, so this microphone icon is always visible:
![image](https://github.com/valinet/ExplorerPatcher/assets/33068114/a4854c3e-a119-4ba9-b202-607e459cd72b)
Can this software be used to hide it?

**Comment by pyrates999** [comment]
No.  That is a setting in the settings app in windows.

**Comment by connerdassen** [comment]
It's not, the only thing I can do is move it into the popup icon tray, I
can't hide it entirely unless if I disable my microphone. I want the icon
gone whilst still being able to use my mic.
.

**Reply by pyrates999** [reply]
that maybe application specific then

**Reply by Imold** [reply]
It's intentional design, the microphone icon is always visible when the microphone is active. It exist for the same reason as the light on your webcam, to alert you that it's active. What you are asking for is a security vulnerability. You just have to put up with a microphone icon in the popup icon tray for the sake of other peoples privacy.

**Reply by pyrates999** [reply]
Please close this then as this will not be implemented.

**Reply by connerdassen** [reply]
@Imold 
I know it's intentional design, but I use VoiceMeeter which means the microphone is active 100% of the time, and therefore the icon is always visible. 
It's an eyesore to me and I'd like it gone, and it's since my microphone is always active it's of no use for security, if malware started using my microphone the icon wouldn't change in any way. 
I'll try looking for other solutions.

**Reply by Imold** [reply]
Maybe not for you but it is for everyone who doesn't always have their microphone active. Which is why you will never find a solution, because protecting other peoples privacy is to important to allow you to remove what you consider an eyesore. If it can be done you will never find that information publicly available, because then Microsoft would find it, and would immediately patch Windows to make it not work.

**Reply by connerdassen** [reply]
I don't see why this would be such a huge priority when secretly screen recording, taking screenshots, reading browser history and browser passwords, logging keypresses and accessing files can be easily achieved through an executable that doesn't even require administrator permissions nor gets picked up by antiviruses... If someone manages to gain the access needed to install software to "secretly" listen in on a microphone then there is already many worse things they'd be able to do.

**Reply by Imold** [reply]
I'm just trying to make you under stand that Microsoft will never allow you to remove it. Also it would be a huge priority because of liability they have to patch any vulnerability they know of, otherwise they can be sued if someone exploits it.

**Reply by connerdassen** [reply]
They may not allow me to remove it because of their own privacy concerns, but I wouldn't call it a "vulnerability", a vulnerability is a flaw in a computer system that weakens the overall security of the device/system, i.e makes it easier to hack / break into. They wouldn't get sued over not having a microphone indicator either

**Reply by Imold** [reply]
Until they knew it could be removed, did nothing, and someone took advantage of that to spy on someone.

**Reply by connerdassen** [reply]
that's still not something they could get sued over, when you use windows you agree to their end user license agreement which limits their liability in case of software issues. Furthermore, they're not getting sued now over the many vulnerabilities and privacy issues that exist within windows either.

**Comment by Raraph84** [comment]
I also use Voicemeeter, and the issue is still not resolved.  
Microsoft won't change anything, but it should be a feature of Explorer Patcher.  
Moreover, many Explorer Patcher users restore the Windows 10 taskbar, which does not have this icon.

**Reply by Imold** [reply]
Explorer Patcher is for restoring old features adding new ones that didn't exist in old versions of windows is outside the scope of this project. Also there is a mic active icon for the Windows 10 taskbar, it's just possible to move it to the popup. If that's good enough for you just use the Windows 10 taskbar.
If you want to remove it from the Windows 11 taskbar, you'll have to,
Install [windhawk](https://windhawk.net/),
Install Windows 11 Taskbar Styler, 
Open the Windows 11 Taskbar Styler mod in Windhawk, 
Go to the "Advanced" tab, Copy `{
  "controlStyles[0].target": "SystemTray.Stack#MainStack",
  "controlStyles[0].styles[0]": "Visibility=Collapsed"
}` to the text box under "Mod settings" and click "Save". 
Note the icon will be completely gone it won't be available in the show more icons popup.

**Reply by m417z** [reply]
A Windhawk user pointed me to this answer, and asked for a simpler solution. Hiding other system tray icons was also requested by several users since Windows 11 became available. Finally, I created a mod which allows to hide various system tray icons via simple options: [Taskbar tray system icon tweaks](https://windhawk.net/mods/taskbar-tray-system-icon-tweaks).

Here's a short demonstration:

https://github.com/user-attachments/assets/3b8f06b5-d6ad-4822-bfb1-6faa86477bdd
